include/seqan3/range/view/pairwise_combine.hpp

```diff
@@ -0,0 +1,597 @@
+        //!\endcond
+        {
+            return std::tie(lhs.first_it, lhs.second_it) <= std::tie(rhs.first_it, rhs.second_it);
+        }
+
+        constexpr friend bool operator>=(iterator_type const & lhs, iterator_type const & rhs)
+            noexcept(noexcept(std::declval<source_iterator_type &>() >= std::declval<source_iterator_type &>()))
+        //!\cond
+            requires std::StrictTotallyOrdered<source_iterator_type>
+        //!\endcond
+        {
+            return std::tie(lhs.first_it, lhs.second_it) >= std::tie(rhs.first_it, rhs.second_it);
+        }
+        //!\}
+
+    private:
+
+        /*!\brief Returns the index for the current iterator position.
+         *
+         * \details
+         *
+         * The pairwise combination can also be seen as a triangular matrix, where given a size n the corresponding
+         * matrix has values in [0, 1], [0, 2], ... , [0, n - 1], [1, 2], ..., [1, n - 1], ... [n - 2, n - 1] and all
+         * other entries are empty. Using this scheme one can use the properties of triangular numbers, where the
+         * diagonal index of this matrix can be computed using triangular roots
+         * (see https://en.wikipedia.org/wiki/Triangular_number). Given these properties, one can compute the matrix
+         * index (i, j) from the linearised matrix index and vice versa.
+         */
+        constexpr size_t to_index() const
+            noexcept(noexcept(std::ranges::distance(std::declval<source_iterator_type &>(),
+                                                    std::declval<source_iterator_type &>())))
+        //!\cond
+            requires std::RandomAccessIterator<source_iterator_type>
+        //!\endcond
+        {
+            size_t src_size = std::ranges::distance(begin_it, end_it);
+            size_t index_i = std::ranges::distance(begin_it, first_it);
+            size_t index_j = std::ranges::distance(begin_it, second_it);
+            return (src_size * (src_size - 1)/2) - (src_size - index_i) * ((src_size - index_i) - 1)/2 +
+                   index_j - index_i - 1;
+        }
+
+        /*!\brief Sets the iterator to the given index.
+         * \param[in] index The index to set the iterator to.
+         * \copydetails to_index()
+         */
+        constexpr void from_index(size_t const index)
+            noexcept(noexcept(std::ranges::distance(std::declval<source_iterator_type &>(),
+                                                    std::declval<source_iterator_type &>())) &&
+                     noexcept(std::ranges::advance(std::declval<source_iterator_type &>(), 1)))
+        //!\cond
+            requires std::RandomAccessIterator<source_iterator_type>
+        //!\endcond
+        {
+            size_t src_size = std::ranges::distance(begin_it, end_it);
+            size_t index_i = src_size - 2 -
+                             std::floor(std::sqrt(-8 * index + 4 * src_size * (src_size - 1) - 7)/2.0 - 0.5);
```

Comment:
pfew.. did you calculate that yourself and can comment this or is there a reference you can link?

Reply:
There is a link to the thread in the details section from the to_index function, which this function copies